client/landing/stepper/declarative-flow/site-migration-flow.ts

```diff
@@ -26,6 +27,36 @@ import type { AssertConditionResult, Flow, ProvidedDependencies } from './intern
 
 const FLOW_NAME = 'site-migration';
 
+/**
+ * Redirects users coming from landing pages and "/sites" to the new migration flow.
+ */
+const useMigrationFlowRevampRedirect = () => {
+	const [ query ] = useSearchParams();
+	const fullQueryString = query.toString();
+	const ref = query.get( 'ref' ) ?? '';
+	const from = query.get( 'from' ) ?? '';
+
+	// Redirect user to the new migration flow.
+	useEffect( () => {
+		if (
+			! config.isEnabled( 'migration-flow/revamp' ) &&
+			! window.localStorage.getItem( 'migration-flow/revamp' )
+		) {
+			return;
+		}
+
+		// If user is coming from 'move-lp' and has a source site, we assume it's a WordPress site because it's already validated there.
+		if ( 'move-lp' === ref && '' !== from ) {
+			window.location.replace(
+				`/setup/migration/${ STEPS.SITE_CREATION_STEP.slug }?${ fullQueryString }`
```

Comment:
We need to include the query param 'signup=1' when we redirect the user to trigger the calypso_signup_start

Reply:
https://github.com/Automattic/wp-calypso/pull/93417/commits/5e9c70670fea6e1f63d528e52561526aa933091b